components/kyma-environment-broker/internal/process/provisioning/secret_cm_overrides.go

```diff
@@ -0,0 +1,123 @@
+package provisioning
+
+import (
+	"context"
+	"time"
+
+	"github.com/kyma-incubator/compass/components/kyma-environment-broker/internal"
+	"github.com/kyma-incubator/compass/components/kyma-environment-broker/internal/process"
+	"github.com/kyma-incubator/compass/components/kyma-environment-broker/internal/ptr"
+	"github.com/kyma-incubator/compass/components/kyma-environment-broker/internal/storage"
+	"github.com/kyma-incubator/compass/components/provisioner/pkg/gqlschema"
+
+	"github.com/sirupsen/logrus"
+	coreV1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	namespace      = "compass-system"
+	overrideLabel  = "provisioning-runtime-override"
+	componentLabel = "component"
+)
+
+type OverridesFromSecretsAndConfigStep struct {
+	ctx              context.Context
+	k8sClient        client.Client
+	operationManager *process.OperationManager
+}
+
+func NewOverridesFromSecretsAndConfigStep(c context.Context, cli client.Client, os storage.Operations) *OverridesFromSecretsAndConfigStep {
+	return &OverridesFromSecretsAndConfigStep{
+		ctx:              c,
+		k8sClient:        cli,
+		operationManager: process.NewOperationManager(os),
+	}
+}
+
+func (s *OverridesFromSecretsAndConfigStep) Name() string {
+	return "Overrides_From_Secrets_And_Config_Step"
+}
+
+func (s *OverridesFromSecretsAndConfigStep) Run(operation internal.ProvisioningOperation, log logrus.FieldLogger) (internal.ProvisioningOperation, time.Duration, error) {
+	overrides := make(map[string][]*gqlschema.ConfigEntryInput, 0)
+	globalOverrides := make([]*gqlschema.ConfigEntryInput, 0)
+
+	secretList := &coreV1.SecretList{}
+	if err := s.k8sClient.List(s.ctx, secretList, s.listOptions()...); err != nil {
+		log.Errorf("cannot fetch list of secrets: %s", err)
+		return operation, 10 * time.Second, nil
```

Comment:
shouldn't we use the `RetryOperation ` method? right now the step will be executed  infinitely?


same for other error handling in this step